lib/evaluation/handlers/sameSubjectEvaluationHandler.ts

```diff
@@ -184,15 +166,13 @@ export class SameSubjectEvaluationHandler extends AbstractEvaluationHandler {
   ) {
     const entry = this.getValue(fieldIdzInputDescriptors, searchableFieldIds);
 
-    searchableFieldIds.forEach((searchableFieldId) => entry.mappedFieldIds.add(searchableFieldId));
+    searchableFieldIds.forEach((searchableFieldId) => entry?.mappedFieldIds.add(searchableFieldId));
 
-    inputDescriptorIds.forEach((inputDescriptorId) => entry.mappedInputDescriptorIds.add(inputDescriptorId));
+    inputDescriptorIds.forEach((inputDescriptorId) => entry?.mappedInputDescriptorIds.add(inputDescriptorId));
   }
 
   private findAllDescribedCredentialsPaths() {
-    if (this.vPresentation.presentation_submission) {
-      this.vPresentation.presentation_submission.descriptor_map.forEach(this.descriptorToPathMapper());
-    }
+    this.vPresentation?.presentation_submission.descriptor_map.forEach(this.descriptorToPathMapper());
```

Comment:
Can presentation_submission be undefined? The change doesn't do the same as the original code, which assumed the vPresentation was always present (current does not), but which checked for the presentation_submission to be defined (current does not)